File: JavaScript/Homeworks/02/feladat.js
// Nev: Barna Gergely
// Neptun: FJKXGG
// h: h144988

function matek(valtozo) {
    if (valtozo == null) {
        return 0;
    } else if (typeof valtozo === "string") {
        return 1;
    } else if (typeof valtozo === "number" && Number.isInteger(valtozo)) {
          let temp;
            if(valtozo === 0) {
              return 1;
            }
            else if (valtozo % 2 === 0) {
                temp = 1;
                for (let i = 0; i < valtozo; i++) {
                    temp = temp*valtozo;
                }
            } else {
                temp = ((valtozo-1)*(valtozo-1))
            }
            return temp;
    } else {
        return null;
    }
}

console.log(matek(10));
console.log(matek("5"));
console.log(matek("alma"));
console.log(matek(5));
console.log(matek(2));
console.log(matek(0));
console.log(matek(1.48));
console.log(matek());

File: Python/Exams/01/feladat.py
# Nev: Barna Gergely
# Neptun: FJKXGG
# h: h144988

def ketszer(parameter):
    if isinstance(parameter, str):
        lista = parameter.split(", ")

        elemek = dict()
        for elem in lista:
            if elemek.get(elem) == None:
                elemek[elem] = 1
            else:
                elemek[elem] = elemek[elem] + 1

        vissza = list()
        for kulcs in elemek:
            if elemek[kulcs] == 2:
                vissza.append(kulcs)
        return vissza

class Temeto:
    def __init__(self, nev, szam = 10):
        self.nev = nev

        if szam <= 10:
            self._sirok_szama = 10
        else:
            self._sirok_szama = szam

        self.szellemek = list()

    @property
    def sirok_szama(self):
        return self._sirok_szama

    @sirok_szama.setter
    def sirok_szama(self, szam):
        if szam <= 10:
            self._sirok_szama = 10
        else:
            self._sirok_szama = szam

    def __str__(self):
        if len(self.szellemek) == 0:
            ertek = "nem kiserti szellem"
        else:
            ertek= str(len(self.szellemek)) + " szellem kiserti"
        return f"A {str(self.nev)} temetoben {str(self._sirok_szama)} sir van, es {ertek}"

    def __iadd__(self, other):
        lista = self.szellemek
        if isinstance(other, str):
            if lista.count(other) == 0:
                lista.append(other)
            else:
                lista[lista.index(other)] = lista[lista.index(other)]+"1"
                lista.append(other+"2")
            return lista
        else:
            raise ValueError("Nem szellem")
    def __eq__(self, other):
        return self.nev == other.nev and self.sirok_szama == other.sirok_szama